app/models/cache/IleQuery.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.cache
+
+import java.time.{Instant, ZoneOffset}
+
+import play.api.libs.json.{JsError, JsNumber, JsObject, JsResult, JsSuccess, JsValue, Json, OFormat}
+
+case class IleQuery(sessionId: String, ucr: String, conversationId: String, createdAt: Instant = Instant.now())
+
+object IleQuery {
+
+  implicit val formatInstant: OFormat[Instant] = new OFormat[Instant] {
+    override def writes(datetime: Instant): JsObject =
+      Json.obj("$date" -> datetime.toEpochMilli)
+
+    override def reads(json: JsValue): JsResult[Instant] =
+      json match {
+        case JsObject(map) if map.contains("$date") =>
+          map("$date") match {
```

Comment:
We can do this in single map lookup i think
```
map.get("$date") match {
  case Some(JsNumber(v)) =>
  case Some(_) =>
  case None =>
}
```

Reply:
In that case we will miss UnexpectedDateFormat. Not sure that's necessary.